src/Inventory/Asset/NetworkEquipment.php

```diff
@@ -343,6 +343,60 @@ public function getStackedSwitches(): array
         return $switches;
     }
 
+    /**
+     * Returns a mapping of port indexes to chassis id
+     *
+     * @return array
+     */
+    public function getStackedPortsMap(): array
+    {
+        $components = $this->extra_data['network_components'] ?? [];
+        foreach ($components as $i => $component) {
+            foreach (['type', 'contained_index', 'index'] as $prop) {
+                if (!property_exists($component, $prop)) {
+                    unset($components[$i]);
+                    continue 2;
+                }
+            }
+        }
+        if (!count($components)) {
+            return [];
+        }
+
+        $map = [];
+        foreach ($components as $component) {
+            if ($component->type == 'port') {
+                $chassis_id = $this->traverseComponents($components, $component->contained_index);
+                if ($chassis_id) {
+                    $map[$component->index] = $chassis_id;
+                }
+            }
+        }
+
+        return $map;
+    }
+
+    private function traverseComponents(array $components, int $index, int $depth = 0): ?int
+    {
+        // max level of recursion
+        if ($depth > 5) {
```

Comment:
Could you give an explanation for this limit? Why a limit? Why set to 5?

Reply:
Just a safeguard from an endless loop. "5" was a blind guess that this level of nesting will suffice.